frontend/packages/console-app/src/actions/providers/project-provider.ts

```diff
@@ -0,0 +1,65 @@
+import { useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useK8sModel } from '@console/dynamic-plugin-sdk/src/lib-core';
+import { useDeleteNamespaceModalLauncher } from '@console/internal/components/modals/delete-namespace-modal';
+import { asAccessReview } from '@console/internal/components/utils/rbac';
+import { K8sModel, K8sResourceKind, referenceFor } from '@console/internal/module/k8s';
+import { CommonActionCreator } from '../hooks/types';
+import { useCommonActions } from '../hooks/useCommonActions';
+
+const useDeleteAction = (kindObj: K8sModel, resource: K8sResourceKind) => {
+  const { t } = useTranslation();
+  const launchDeleteModal = useDeleteNamespaceModalLauncher({ kind: kindObj, resource });
+  const hidden = resource.metadata.name === 'default' || resource.status?.phase === 'Terminating';
+
+  const factory = useMemo(
+    () => ({
+      delete: () => ({
+        id: 'delete-project',
+        label: t('console-app~Delete {{label}}', { label: t(kindObj.labelKey) }),
+        cta: launchDeleteModal,
+        accessReview: asAccessReview(kindObj, resource, 'delete'),
+      }),
+    }),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
can we get some explanation for why this is disabled in a line above the `eslint-disable-next-line`?